### PR TITLE
[Enhancement](Load) stream Load using SQL

### PR DESCRIPTION
**Pull request by zzzzzzzs**
## Proposed changes

Using stream load in SQL mode

for example:
example.csv
```
10000,北京
10001,天津
```

``` curl
curl -v --location-trusted -u root: -H "sql: insert into test.t1(c1, c2) select c1,c2 from stream(\"format\" = \"CSV\", \"column_separator\" = \",\")" -T example.csv http://127.0.0.1:8030/api/_stream_load_with_sql
```

``` curl
curl -v --location-trusted -u root: -H "sql: insert into test.t2(c1, c2, c3) select c1,c2, 'aaa' from stream(\"format\" = \"CSV\", \"column_separator\" = \",\")" -T example.csv http://127.0.0.1:8030/api/_stream_load_with_sql
```

``` curl
curl -v --location-trusted -u root: -H "sql: insert into test.t3(c1, c2) select c1, count(1) from stream(\"format\" = \"CSV\", \"column_separator\" = \",\") group by c1" -T example.csv http://127.0.0.1:8030/api/_stream_load_with_sql
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java [hunk 1]
```diff
@@ -255,14 +255,34 @@ public void createScanRangeLocations() throws UserException {
             ConnectContext.get().getExecutor().getSummaryProfile().setGetSplitsFinishTime();
         }
         this.inputSplitsNum = inputSplits.size();
-        if (inputSplits.isEmpty()) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java:258]: why delete this line?

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java [hunk 2]
```diff
@@ -469,6 +489,8 @@ protected static Optional<TFileType> getTFileType(String location) {
                 return Optional.of(TFileType.FILE_LOCAL);
             } else if (location.startsWith(FeConstants.FS_PREFIX_OFS)) {
                 return Optional.of(TFileType.FILE_BROKER);
+            } else if (location.startsWith(FeConstants.FS_PREFIX_COSN)) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java:492]: why add these code?

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java [hunk 3]
```diff
@@ -256,14 +256,34 @@ public void createScanRangeLocations() throws UserException {
             ConnectContext.get().getExecutor().getSummaryProfile().setGetSplitsFinishTime();
         }
         this.inputSplitsNum = inputSplits.size();
-        if (inputSplits.isEmpty()) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java:259]: should check it is file stream

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 4]
```diff
@@ -67,19 +66,16 @@
...
 
     // set this id before analyze
     protected volatile long stmtId;
     protected volatile long forwardedStmtId;
 
-    // set for stream load with sql
-    protected volatile TUniqueId loadId;
+    // set for http_stream
     protected volatile long backendId;
-    protected volatile LoadTaskInfo streamLoadInfo;
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:82]: do not format the code in this PR. May be in other PR.

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 5]
```diff
@@ -562,6 +562,33 @@ private void traceInstance() {
         }
     }
 
+    private void performFragmentProcessing() throws Exception {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:565]: rename to  processFragmentAssignmentAndParams

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 6]
```diff
@@ -1953,7 +1975,7 @@ private void computeColocateJoinInstanceParam(PlanFragmentId fragmentId,
                     = findOrInsert(assignment, addressScanRange.getKey(), new HashMap<>());
             int expectedInstanceNum = 1;
             if (parallelExecInstanceNum > 1) {
-                //the scan instance num should not larger than the tablets num
+                // the scan instance num should not larger than the tablets num
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1978]: do not do formatting work.

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run  buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.65 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17161822932 Bytes

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.54 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161824947 Bytes

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.87 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17162117854 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**